ckan/logic/schema.py

```diff
@@ -411,6 +411,7 @@ def default_user_schema():
         'reset_key': [ignore],
         'activity_streams_email_notifications': [ignore_missing],
         'state': [ignore_missing],
+        'extras': [ignore_missing, dict]
```

Comment:
The JSON is already validated when the POST request is parsed.

It's not a string, it's a literal JSON dict, deserialized into a python dict by the normal `json.loads()` in the API. Checking `startswith('{')` wouldn't be sufficient anyways.